src/OpenTelemetry.Collector.Dependencies/AzureSdkDiagnosticListener.cs

```diff
@@ -60,6 +63,13 @@ public override void OnStartActivity(Activity current, object valueValue)
                 .SetCreateChild(false)
                 .SetSampler(this.sampler);
 
+            var links = LinksPropertyFetcher.Fetch(valueValue) as IEnumerable<Activity> ?? Array.Empty<Activity>();
+
+            foreach (var link in links)
+            {
+                spanBuilder.AddLink(Link.FromActivity(link));
+            }
+
             spanBuilder.SetSpanKind(isHttp ? SpanKind.Client : SpanKind.Internal);
```

Comment:
this doesn't look right, it should be based on tag

Reply:
done